src/modules/esl-share/core/esl-share-button.ts

```diff
@@ -0,0 +1,101 @@
+import {ESLBaseElement} from '../../esl-base-element/core';
+import {attr, boolAttr, listen} from '../../esl-utils/decorators';
+import {ENTER, SPACE} from '../../esl-utils/dom/keys';
+import {ESLShareActionRegistry} from './esl-share-action-registry';
+
+import type {ESLShareList} from './esl-share-list';
+
+export interface ShareButtonConfig {
+  'action': string;
+  'id': string;
+  'icon': string;
+  'iconBackground': string;
+  'link': string;
+  'name': string;
+  'title': string;
+}
+
+export class ESLShareButton extends ESLBaseElement {
+  public static override is = 'esl-share-button';
+
+  public static DEFAULT_ICON_BG_COLOR: string = '#FFF';
+
+  @attr() public action: string;
+  @attr() public buttonId: string;
+  @attr() public link: string;
+  @attr() public name: string;
+  @boolAttr() public unavailable: boolean;
+
+  public static build(cfg: ShareButtonConfig): ESLShareButton | null {
+    const shareAction = ESLShareActionRegistry.instance.get(cfg.action);
+    if (!shareAction) return null;
+
+    const {isAvailable} = shareAction;
+    const $button = ESLShareButton.create();
+    $button.$$attr('action', cfg.action);
+    $button.$$attr('button-id', cfg.id);
```

Comment:
Discussion 02.03
Assumption:  redundant as uses for the same purpose as `action`